packages/web/src/components/search/DataSearch.js

```diff
@@ -1288,7 +1314,10 @@ const mapStateToProps = (state, props) => ({
 	total: state.hits[props.componentId] && state.hits[props.componentId].total,
 	hidden: state.hits[props.componentId] && state.hits[props.componentId].hidden,
 	popularSuggestions: state.querySuggestions[props.componentId],
-	defaultPopularSuggestions: state.defaultPopularSuggestions[props.componentId],
+	defaultPopularSuggestions: (state.defaultPopularSuggestions[props.componentId] || [])
```

Comment:
@anik-ghosh-au7 Why are you doing the calculation in `mapStateToProps`? It would cause unnecessary re-rendering. I mentioned the place where you should calculate it.